src/org-manager/components/new-pbas-confirm/new-pbas-confirm.component.ts

```diff
@@ -12,7 +15,53 @@ export class NewPBAsConfirmComponent {
   @Input() public formControls: FormControl[];
   @Input() public newPBAs: Map<string, string>;
 
-  constructor() {
-
+  constructor(
+    private readonly router: Router,
+    private readonly pbaService: PbaService
+  ) {
   }
+
+  public confirmPBAs(): void {
+    const pbaNumbers = [];
+
+    this.newPBAs.forEach((value: string, key: string) => {
+      pbaNumbers.push({
+        pbaNumber: key,
+        status: value,
+        statusMessage: ''
+      })
+    });
+
+    const payload = {
+      pbaNumbers,
+      orgId: this.org.organisationId
+    }
+
+    this.pbaService.setPBAStatus(payload)
+      .subscribe({
```

Comment:
should we not do any subscription clean up as well?

Reply:
Cleanup complete - I've added a subscription property that is unsubscribed in the ngOnDestroy method.